enumerables.rb

```diff
@@ -0,0 +1,127 @@
+module Enumerable
+  def my_each
+    if block_given?
+      array = to_a
+      array.size.times { |index| yield array[index] }
+      return self
+    end
+    to_enum
+  end
+
+  def my_each_with_index
+    if block_given?
+      array = to_a
+      array.size.times { |index| yield(array[index], index) }
+      return self
+    end
+    to_enum
+  end
+
+  def my_select
+    if block_given?
+      result = []
+      my_each { |item| result << item if yield item }
+      result
+    end
+    to_enum
+  end
+
+  def my_all?(pattern = nil, &block)
+    if block_given? || pattern.nil?
+      helper = block_given? ? block : proc { |value| value }
+      my_each { |value| return false unless helper.call(value) }
+    else
+      my_each { |value| return false unless match_pattern?(value, pattern) }
+    end
+
+    true
+  end
+
+  def my_any?(pattern = nil)
+    if block_given? || pattern.nil?
+      helper = block_given? ? block : proc { |value| value }
```

Comment:
### my_any

- raises an error when a block is passed `undefined local variable or method 'block' for Array`, 
make sure my_any? method expects a block in its parameters.